### PR TITLE
new infra updates

### PR DESCRIPTION
**Pull request by whd**
WIP

### CODE REVIEW

**Comment by whd** [comment]
The extract_dimensions module is supposed to work with both old and new landfill types, but I've only tested it with new. The geoip config should probably be optional since if we're processing old style landfill we won't need to do the lookups.

File: moz_telemetry/modules/moz_telemetry/extract_dimensions.lua [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+local mtn    = require "moz_telemetry.normalize"
+local dt     = require "lpeg.date_time"
+local geoip  = require "geoip.city"
+
+local read_config          = read_config
+local assert               = assert
+local pairs                = pairs
+local ipairs               = ipairs
+local create_stream_reader = create_stream_reader
+local inject_message	   = inject_message
```
Comment (trink) [inline moz_telemetry/modules/moz_telemetry/extract_dimensions.lua]: alignment

**Comment by trink** [comment]
bump the package version to 1.0.2

File: moz_telemetry/sandboxes/heka/analysis/moz_telemetry_payload_size.lua [hunk 2]
```diff
@@ -0,0 +1,46 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
```
Comment (trink) [inline moz_telemetry/sandboxes/heka/analysis/moz_telemetry_payload_size.lua]: Needs an H1 entry

File: moz_telemetry/modules/moz_telemetry/extract_dimensions.lua [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+            local ok, err = pcall(inject_message, msg)
+            if not ok then
+                emsg.Fields.DecodeErrorType = "inject_message"
+                emsg.Fields.DecodeError = err
+                pcall(inject_message, emsg)
+            end
+        end
+    end
+
+    return 0
```
Comment (trink) [inline moz_telemetry/modules/moz_telemetry/extract_dimensions.lua]: Don't need the return value

File: moz_telemetry/sandboxes/heka/analysis/moz_telemetry_payload_size.lua [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        submissionDate = "",
+    }
+}
+
+function process_message()
+    msg.Timestamp = read_message("Timestamp")
+    msg.Fields.build = read_message("Fields[appBuildId]")
+    msg.Fields.channel = read_message("Fields[appUpdateChannel]")
+    msg.Fields.docType = read_message("Fields[docType]")
+    msg.Fields.size = read_message("Fields[Size]")
```
Comment (trink) [inline moz_telemetry/sandboxes/heka/analysis/moz_telemetry_payload_size.lua]: Can use read_message("size").

File: moz_telemetry/sandboxes/heka/input/moz_telemetry_heka_kafka.lua [hunk 5]
```diff
@@ -0,0 +1,73 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (trink) [inline moz_telemetry/sandboxes/heka/input/moz_telemetry_heka_kafka.lua]: This seems like the kafka/sandboxes/heka/input/heka_kafka.lua should be generalized to accepted a decoder and if nil uses the existing behavior (pass through injection).  To keep the terminology the same as Heka it should be a 'decoder' module.

File: moz_telemetry/sandboxes/heka/output/moz_telemetry_s3.lua [hunk 6]
```diff
@@ -69,6 +75,11 @@ if compression and compression ~= "zst" and compression ~= "gz" then
     error("compression must be nil, zst or gz")
 end
 
+local storage_class         = read_config("storage_class") or "STANDARD"
+if storage_class and storage_class ~= "STANDARD" and
```
Comment (trink) [inline moz_telemetry/sandboxes/heka/output/moz_telemetry_s3.lua]: don't need the extra storage_class check here

File: moz_telemetry/sandboxes/heka/output/moz_telemetry_s3.lua [hunk 7]
```diff
@@ -153,6 +164,8 @@ local function get_entry(path)
 end
 
 local dimensions = mts3.validate_dimensions(read_config("dimension_file"))
+-- create the batch directory if it does not exist
+os.execute(string.format("mkdir -p %s", batch_dir))
```
Comment (trink) [inline moz_telemetry/sandboxes/heka/output/moz_telemetry_s3.lua]: probably want to test the return code

File: moz_telemetry/sandboxes/heka/output/moz_telemetry_s3_webrtc.lua [hunk 8]
```diff
@@ -0,0 +1,240 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (trink) [inline moz_telemetry/sandboxes/heka/output/moz_telemetry_s3_webrtc.lua]: duplicates too much code with moz_telemetry_s3

File: moz_telemetry/modules/moz_telemetry/webrtc.lua [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    if type(payload) ~= "table" then return false end
+    local w = payload["webrtc"] or {}
+    local i = w["IceCandidatesStats"] or {}
+    if next(i["webrtc"] or {}) or next(i["loop"] or {}) then
+        return true
+    end
+    return false
+end
+
+function filter_message ()
```
Comment (trink) [inline moz_telemetry/modules/moz_telemetry/webrtc.lua]: Naming is a bit counter intuitive.  i.e. `if filter_message()`  to me true implies that the message should be filtered out (not kept)

File: moz_telemetry/modules/moz_telemetry/extract_dimensions.lua [hunk 10]
```diff
@@ -60,7 +60,11 @@ local schema_path   = read_config("schema_path") or error("schema_path must be s
 local content_field = read_config("content_field") or "Fields[content]"
 local uri_field = read_config("content_field") or "Fields[uri]"
 
-local city_db = assert(geoip.open(read_config("geoip_city_db")))
+local city_db_file = read_config("geoip_city_db")
```
Comment (trink) [inline moz_telemetry/modules/moz_telemetry/extract_dimensions.lua]: You can actually make the inclusion of the entire module optional.  i.e. if you aren't using geoip decoding you don't even have to have load the `.so`

**Comment by trink** [comment]
Lets formally document the splitter?, filter?, decoder, and encoder APIs recommendations (I don't want you to have to modify this code again).   Currently we have something basic like this:

```lua
-- xxx_decoder_module.lua
function decode(string) -- turns the string into something consumable by inject_message
  -- transform string
  if not filter() then
inject_message 
  return nil
end

-- xxx_encoder_module.lua
function encode() -- no parameters as it works on the current message
  -- transform message
  return string or nil (filtered) or throw error
end
```

Splitters: we haven't addressed record separation at all and it will be very interesting in Lua (a good starting place may be to look at  http://w3.impa.br/~diego/software/luasocket/ltn12.html)

Filtering: in many cases it could just be part of the decoder/encoder but in some cases you will want it to run before (like the webrtc output)

@rafrombrc

**Pull request merged**